src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,19 +33,27 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={query}
+          onChange={(event) => dispatch(actions.filterByQuery(event.target.value))}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
-        <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
-        </span>
+        {query && (
+          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
+            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+            <button
+              data-cy="clearSearchButton"
+              type="button"
+              className="delete"
+              onClick={() => {
+                dispatch(actions.filterByQuery(''));
+                dispatch(actions.filterByStatus('all'));
+              }}
+            />
```

Comment:
Why disable this? Just add `aria-label="Clear search"` attribute to button for screenreaders.
```suggestion
          <span className="icon is-right" style={{ pointerEvents: 'all' }}>
            <button
              aria-label="Clear search"
              data-cy="clearSearchButton"
              type="button"
              className="delete"
              onClick={() => {
                dispatch(actions.filterByQuery(''));
                dispatch(actions.filterByStatus('all'));
              }}
            />
```